src/query/plan/preprocess.hpp

```diff
@@ -79,6 +99,12 @@ class UsedSymbolsCollector : public HierarchicalTreeVisitor {
 
   std::unordered_set<Symbol> symbols_;
   const SymbolTable &symbol_table_;
+
+ private:
+  struct Scope {
+    bool in_exists{false};
+  };
+  std::vector<Scope> scopes_;
```

Comment:
You never change the size of `scopes_` from 1, so maybe they are not needed. A single `bool in_exists_` shuld suffice in my opinion.

Reply:
Agree